serviceconnector/src/createLinker/KeyVaultListStep.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzExtPipelineResponse, sendRequestWithTimeout } from "@microsoft/vscode-azext-azureutils";
+import { AzureWizardPromptStep, IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import * as vscode from 'vscode';
+import { ICreateLinkerContext } from "./ICreateLinkerContext";
+
+export interface KeyVaultJsonResponse {
+    value: KeyVaultAccountJsonResponse[];
+}
+
+export interface KeyVaultAccountJsonResponse {
+    id: string;
+    name: string;
+}
+
+export class KeyVaultListStep extends AzureWizardPromptStep<ICreateLinkerContext> {
+    public async prompt(context: ICreateLinkerContext): Promise<void> {
+        const placeHolder: string = vscode.l10n.t('Select a Key Vault');
+        const accounts = await getKeyVaultAccounts(context);
+        context.keyVaultAccount = (await context.ui.showQuickPick(this.getPicks(accounts.value), { placeHolder })).data;
+    }
+
+    public shouldPrompt(context: ICreateLinkerContext): boolean {
+        return !context.keyVaultAccount;
+    }
+
+    private async getPicks(accounts: KeyVaultAccountJsonResponse[]): Promise<IAzureQuickPickItem<KeyVaultAccountJsonResponse>[]> {
+        return accounts.map(d => {
+            return { label: d.name, data: d };
+        });
+    }
+}
+
+async function getKeyVaultAccounts(context: ICreateLinkerContext): Promise<KeyVaultJsonResponse> {
+    const url = `https://management.azure.com/subscriptions/${context.subscriptionId}/resources?$filter=resourceType eq 'Microsoft.KeyVault/vaults'&api-version=2015-11-01`
```

Comment:
We don't want to use `management.azure.com` for these requests because then we don't support anything but the public cloud. I think if you check the `context.subscription` or maybe `context.credentials` you should see there's something called a `resourceManagementEndpoint` or something like that,